src/amoc_telemetry.erl

```diff
@@ -2,14 +2,42 @@
 %% @copyright 2023 Erlang Solutions Ltd.
 -module(amoc_telemetry).
 
--export([execute/3]).
+-export([execute/3, execute_error/3, execute_msg_to_self/3, execute_nodes/3]).
 
 -spec execute(EventName, Measurements, Metadata) -> ok when
       EventName :: telemetry:event_name(),
       Measurements :: telemetry:event_measurements(),
       Metadata :: telemetry:event_metadata().
 execute(Name, Measurements, Metadata) ->
     TimeStamp = erlang:monotonic_time(),
-    NameWithAmocPrefix = [amoc | Name],
+    PrefixedName = [amoc | Name],
     MetadataWithTS = Metadata#{monotonic_time => TimeStamp},
-    telemetry:execute(NameWithAmocPrefix, Measurements, MetadataWithTS).
+    telemetry:execute(PrefixedName, Measurements, MetadataWithTS).
+
+-spec execute_error(EventName, Metadata, Msg) -> ok when
+      EventName :: telemetry:event_name(),
+      Metadata :: telemetry:event_metadata(),
+      Msg :: binary().
+execute_error(Name, Metadata, Message) ->
+    PrefixedName = [amoc | Name],
+    MetadataWithLog = Metadata#{log_level => error, msg => Message},
+    telemetry:execute(PrefixedName, #{error => 1}, MetadataWithLog).
+
+-spec execute_msg_to_self(EventName, Metadata, Msg) -> ok when
+      EventName :: telemetry:event_name(),
+      Metadata :: telemetry:event_metadata(),
+      Msg :: binary().
+execute_msg_to_self(Name, Metadata, Message) ->
```

Comment:
execute_msg_to_self is absolutely horible name, it's external API but it means absolutely nothing for developers. also such interface should be using `amoc_telemetry:execute/3` and, since it's used in just one module, it should be placed in the `amoc_throttle_proces` module.